pkg/rpc/rpc.go

```diff
@@ -0,0 +1,61 @@
+package rpc
+
+import (
+	"fmt"
+
+	rpcclient "github.com/cometbft/cometbft/rpc/client/http"
+	"github.com/cosmos/cosmos-sdk/client"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"google.golang.org/grpc"
+
+	authoritytypes "github.com/zeta-chain/zetacore/x/authority/types"
+	crosschaintypes "github.com/zeta-chain/zetacore/x/crosschain/types"
+	fungibletypes "github.com/zeta-chain/zetacore/x/fungible/types"
+	lightclienttypes "github.com/zeta-chain/zetacore/x/lightclient/types"
+	observertypes "github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+// Clients contains RPC client interfaces to interact with zetacored
+type Clients struct {
+	AuthorityClient authoritytypes.QueryClient
+	CctxClient      crosschaintypes.QueryClient
+	FungibleClient  fungibletypes.QueryClient
+	AuthClient      authtypes.QueryClient
+	BankClient      banktypes.QueryClient
+	ObserverClient  observertypes.QueryClient
+	LightClient     lightclienttypes.QueryClient
+}
+
+func newClients(ctx client.Context) (Clients, error) {
+	return Clients{
+		AuthorityClient: authoritytypes.NewQueryClient(ctx),
+		CctxClient:      crosschaintypes.NewQueryClient(ctx),
+		FungibleClient:  fungibletypes.NewQueryClient(ctx),
+		AuthClient:      authtypes.NewQueryClient(ctx),
+		BankClient:      banktypes.NewQueryClient(ctx),
+		ObserverClient:  observertypes.NewQueryClient(ctx),
+		LightClient:     lightclienttypes.NewQueryClient(ctx),
+	}, nil
+}
+
+// NewCometBFTClients creates a Clients which uses cometbft abci_query as the transport
+func NewCometBFTClients(url string) (Clients, error) {
+	cometRPCClient, err := rpcclient.New(url, "/websocket")
```

Comment:
Should be use HTTP instead?

Reply:
The transport is HTTP. It's using the cometbft rpc wrapper though.

Reply:
Yeah this is the only client that satisfies the client parameter that is required: `func (ctx client.Context) WithClient(client client.TendermintRPC) client.Context`. Reviewing the imported code, the websocket is not opened unless you do an operation that requires it. Which should never happen since we do not expose the client directly.